node/src/command.rs

```diff
@@ -322,22 +322,57 @@ pub fn run() -> Result<()> {
 					}),
 				},
 				#[cfg(not(feature = "runtime-benchmarks"))]
-				BenchmarkCmd::Storage(_) =>
+				(BenchmarkCmd::Storage(_), _) =>
 					return Err(sc_cli::Error::Input(
 						"Compile with --features=runtime-benchmarks \
 						to enable storage benchmarks."
 							.into(),
 					)
 					.into()),
 				#[cfg(feature = "runtime-benchmarks")]
-				BenchmarkCmd::Storage(cmd) => runner.sync_run(|config| {
-					let partials = new_partial(&config)?;
-					let db = partials.backend.expose_db();
-					let storage = partials.backend.expose_storage();
-
-					cmd.run(config, partials.client.clone(), db, storage)
-				}),
-				BenchmarkCmd::Machine(cmd) =>
+				(BenchmarkCmd::Storage(cmd), runtime) => match runtime {
```

Comment:
What is the reason that we need to have this new code duplication?

Reply:
It is to get the benchmark running. The `new_partials` have generics, and it wasn't defined. Running the _`-features runtime-benchmarks`_ failed because the first type it was looking for is `Amplitude`, even though the runtime is `Pendulum`